packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -466,9 +466,15 @@ function DashboardLayout(props: DashboardLayoutProps) {
       >
         {drawerContent}
       </Drawer>
-      <Box component="main" sx={{ flexGrow: 1 }}>
+      <Box
+        component="main"
+        sx={{
+          flexGrow: 1,
+          minWidth: { xs: isMobileNavigationOpen ? '100vw' : 'auto', md: 'auto' },
```

Comment:
Let's add a comment to the github issue on mui core repository that tracks the issues with the responsive drawer demo.

Reply:
Raised the issue in https://github.com/mui/material-ui/issues/43244 and added todo comments in https://github.com/mui/mui-toolpad/pull/3905/commits/5658dc02bbb7ddbdae6a513e1ad1fa8813d0521e